indexes/utxoindex/src/index.rs

```diff
@@ -29,8 +29,12 @@ pub struct UtxoIndex {
 
 impl UtxoIndex {
     /// Creates a new [`UtxoIndex`] within a [`RwLock`]
-    pub fn new(consensus: DynConsensus, db: Arc<DB>) -> RwLock<Self> {
-        RwLock::new(Self { consensus, store: Store::new(db) })
+    pub fn new(consensus: DynConsensus, db: Arc<DB>) -> UtxoIndexResult<Arc<RwLock<Self>>> {
+        let mut utxoindex = Self { consensus, store: Store::new(db) };
+        if !utxoindex.is_synced()? {
```

Comment:
Note: this is also one of the reasons why my native utxoindex test will fail with this pr, since the utxoindex is created with a consensus that does not call init() before it is passed.

Reply:
This was fixed by processing genesis just when constructing consensus